cl/lib/search_utils.py

```diff
@@ -1031,6 +1035,36 @@ def add_depth_counts(
         return None
 
 
+def get_parenthetical_groups_with_cache(
+    cluster: OpinionCluster,
+) -> List[ParentheticalGroup]:
+    """
+    Given an opinion cluster, get its parentheticals and categorize them
+    into groups based on textual similarity.
+
+    @param cluster: The cluster to fetch and group parentheticals for
+    @return: A list of ParentheticalGroup's for the given cluster
+    """
+
+    cache_key = f"parenthetical_groups:{cluster.pk}"
+    cache = caches["db_cache"]
+    cached_groups, num_parentheticals = cache.get(cache_key, (None, None))
+    if (
+        cached_groups is not None
+        # Groups will be recomputed when new parentheticals are added
+        and num_parentheticals == cluster.parentheticals.count()
+    ):
+        return cached_groups
+
+    # If the case has more than 1,000 parentheticals, only use the top 1,000
+    parentheticals = list(cluster.parentheticals[:1000])
+    groups = get_parenthetical_groups(parentheticals)
+    a_week = 60 * 60 * 24 * 7
+    cache.set(cache_key, (groups, len(parentheticals)), a_week)
```

Comment:
So I think it's right to use Django's DB cache here since these groups will take up space. I appreciate that decision. 

But I'm actually not sure there's much point in expiring the cache at all. The funny thing about django's DB cache is that items are only culled from it when you run `cache.get` on them. If you never do that, items never get culled because, as you probably know, relational databases like postgresql don't have ttl primitives built into them. (Using an in-memory DB like redis is another story.)

So put another way, once you cache something in the DB, if you have a get-check-expired-set-again pattern like above, that key is going to take up space in the DB forever. It'll take up space when you set it the first time, then it'll still be there after it expires, and someday when you learn it expired (by doing a `get`), the very next thing you do after the `get` culls it is set it again with fresh data.

In my mind, the only reasons to clear a cache are:
 - The data is old (but we check for this via parentheticals.count); or
 - To save space in our cache (but we never get this space back as explained above).

The only other reason we might want to expire it is if our scoring algo changes, but I think I'd argue that we'd want to clear the cache at that point anyway.

So with all that in mind, let's just not expire them unless y'all can think of another reason to. Or if we must, let's set really long ttls of a month or two?

Reply:
Yeah, good call. I added the count checking after the caching, and it didn't occur to me that that obviated the need for expiring the cache.